wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -384,6 +387,54 @@ public final class ZMUserSession: NSObject {
         self.contextStorage = contextStorage
         self.recurringActionService = recurringActionService
         self.dependencies = dependencies
+
+        super.init()
+
+        setupAnalyticsSession()
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleAppDidBecomeActive),
+            name: UIApplication.didBecomeActiveNotification,
+            object: nil
+        )
```

Comment:
Rather than observing notifications here, the session manager could inform the user session when the app became active (by calling a method `applicationDidBecomActive` or something like that.

Reply:
Any thoughts about this @agisilaos ?

Reply:
Done: https://github.com/wireapp/wire-ios/pull/1618/commits/8d296f92daaf0ba9d76c4e2d1f3e6c648605fddc